crates/iceberg/src/scan.rs

```diff
@@ -163,6 +178,54 @@ impl TableScan {
 
         Ok(iter(file_scan_tasks).boxed())
     }
+
+    /// Transforms a stream of FileScanTasks from plan_files into a stream of
+    /// Arrow RecordBatches.
+    pub fn open(&self, mut tasks: FileScanTaskStream) -> crate::Result<ArrowRecordBatchStream> {
+        let file_io = self.file_io.clone();
+        let batch_size = self.batch_size.unwrap_or(DEFAULT_BATCH_SIZE);
+        let projection_mask = self.get_arrow_projection_mask();
+        let row_selection = self.get_arrow_row_selection();
+
+        Ok(
+            try_stream! {
+                while let Some(Ok(task)) = tasks.next().await {
+                    let parquet_reader = file_io
+                        .new_input(task.data_file().file_path())?
+                        .reader()
+                        .await?;
+
+                    let mut batch_stream = ParquetRecordBatchStreamBuilder::new(parquet_reader)
+                        .await
+                        .map_err(|err| {
+                            Error::new(ErrorKind::Unexpected, "failed to load parquet file").with_source(err)
```

Comment:
```suggestion
                            Error::new(ErrorKind::Unexpected, "Failed to load parquet file").with_source(err).with_context("filename", "<input file path>")
```
Please also add context here better error reporting.